module/zfs/spa_stats.c

```diff
@@ -153,9 +152,7 @@ spa_read_history_init(spa_t *spa)
 	ssh->size = 0;
 	ssh->private = NULL;
 
-	(void) snprintf(name, KSTAT_STRLEN, "zfs/%s", spa_name(spa));
-
-	ksp = kstat_create(name, 0, "reads", "misc",
+	ksp = kstat_create("", 0, "reads", "misc",
```

Comment:
There are only a half dozen or so places where a KSTAT_STRLEN sized buffer is put on the stack. In all those cases it's used as part of a `snprintf()` to print a custom name.  These few instances could be easily switched to use `kmem_asprintf()` which would further reduce our stack usage and allow us to make the minimal change of increasing the size of KSTAT_STRLEN.